Tapjoy/tpe_prebid_service
# [NGS-323] TaurusX MRAID updates

## JIRA
https://tapjoy.atlassian.net/browse/NGS-323

## Description
This PR is to update MRAID requests for TaurusX.

All other partners are asking for a banner object to response with MRAID. But TaurusX told us that they're respecting to the value of Imp.Video.API (which needs to include 3 & 5 for MRAID). With this change, when a request is MRAID Supported, then we append Imp.Banner.API to Imp.Video.API and remove Banner object in the request.

We'll keep monitoring if we can get an MRAID response back from TaurusX from here: https://app.signalfx.com/#/temp/chart/EyzQbvsAcAA

This PR also has a small update on Rubicon adapter, which is deep copying the request object. Otherwise any change or update on the openrtb.BidRequest object is reflected to the other adapters. In this specific case, Rubicon's companion ad details are included in the adapters that are built after Rubicon adapter. Other adapters are not updating pointer objects in the main openrtb.BidRequest, so for now only Rubicon change is ok.

## How to Test
- Pull the latest branch
- Update MBS to return MRAID supported always
https://github.com/Tapjoy/mediation_bid_service/blob/d1a39a73d5fd33a75a9e10bb4095e05cbd47af1a/src/config/client/prebiddsp/dsp/taurusx.go#L83-L84
- Add debug log statement in tpe_prebid_service `exchange/bidder.go`, just before sending request
```
        // Save a copy of this request for debugging.
	requestDump, err := httputil.DumpRequest(httpReq, true)
	if err != nil {
		fmt.Println(err)
	}
	fmt.Println(string(requestDump))

	httpResp, err := ctxhttp.Do(ctx, bidder.Client, httpReq)
```
- Spin up k8s pods (MBS, tpe_prebid_service, TJS and PS)
- Mock tpe_prebid_service
- Fire an MBS request
- Beautify each DSP response in tpe_prebid_service logs and confirm if TaurusX doesn't have Banner object, and Imp.Video.API has 3,5
